lib/spack/spack/database.py

```diff
@@ -1529,6 +1529,24 @@ def unused_specs(self):
 
         return unused
 
+    def update_explicit(self, spec, explicit):
+        """
+        Update the spec's explicit state in the database.
+
+        Args:
+            spec (Spec): the spec whose install record is being updated
+            explicit (bool): ``True`` if the package was requested explicitly
+                by the user, ``False`` if it was pulled in as a dependency of
+                an explicit package.
+        """
+        rec = self.get_record(spec)
+        if explicit != rec.explicit:
+            with self.write_transaction():
+                message = '{s.name}@{s.version} : marking the package {0}'
+                status = 'explicit' if explicit else 'implicit'
+                tty.msg(message.format(status, s=spec))
```

Comment:
I think this should probably be a debug message. Spack is quite verbose, and I think we should work on limiting what Spack prints by default.

Reply:
Sorry for the delay. This is a debug message now. The only downside I can see it that `spack mark ...` is now completely silent, which might be surprising for users.